objects/src/accounts/code/mod.rs

```diff
@@ -50,17 +50,25 @@ impl AccountCode {
     /// All procedures exported from the provided library will become members of the account's
     /// public interface.
     ///
+    /// # Notes
+    /// - A `faucet` flag has been temporarly added to this procedure enabling correct storage
+    ///   offset of faucet procedures. This is needed because of the faucet reserved storage slot at
+    ///   location 0.
+    ///
     /// # Errors
-    /// Returns an error if the number of procedures exported from the provided library is smaller
-    /// than 1 or greater than 256.
-    pub fn new(library: Library) -> Result<Self, AccountError> {
+    /// - If the number of procedures exported from the provided library is smaller than 1 or
+    ///   greater than 256.
+    pub fn new(library: Library, is_faucet: bool) -> Result<Self, AccountError> {
         // extract procedure information from the library exports
         // TODO: currently, offsets for all procedures are set to 0; instead they should be read
         // from the Library metadata
         let mut procedures: Vec<AccountProcedureInfo> = Vec::new();
+        // TODO: Remove this check. We offset all faucet procedures
+        // by 1 and other account procedures by 0.
```

Comment:
nit: I would probably combine this `TODO` with the `TODO` on lines 63 - 64 because they have the same underlying cause.